controllers/argocd/status.go

```diff
@@ -93,6 +98,21 @@ func (r *ReconcileArgoCD) reconcileStatusDex(cr *argoprojv1a1.ArgoCD) error {
 	return nil
 }
 
+// reconcileStatusSsoConfig will ensure that the SsoConfig status is updated for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileStatusSsoConfig(cr *argoprojv1a1.ArgoCD) error {
+	status := "Unknown"
```

Comment:
@iam-veeramalla @ciiay 

Is my understanding correct that in case there is no conflict, then user might see `dex/keycloak` status as available/running and `ssoConfig` as unknown at the same time?
If so, that could be a little confusing to the user

Reply:
Hi @jaideepr97 , thanks for the comment. We only have `dex` field in the `status` before this change. `ssoConfig` was meant to tell user the error of misconfiguration of multiple sso providers. I thought about the same thing, but default value for all status fields are set to `Unknown` and I chose to make it consistent. Do you think another field name like `ssoMisconfig` would be better? Or change the default value to something else to indicate no error for `ssoConfig`?

Reply:
@ciiay Did we improve this behavior ?

Reply:
@iam-veeramalla Thanks for the review. I didn't get what exact you mean but this function is updated with new status values of `Unknown/Success/Failed` and the condition of each value is as following.
`// Success: Only one SSO provider is configured in CR.`
`// Failed: More than one SSO providers are configure in CR.`
`// Unknown: For some reason the SSO configuration could not be obtained.`

Reply:
This is updated as we discussed, thanks for the review.